test/lib/eventFactory.test.js

```diff
@@ -440,6 +440,100 @@ describe('Event Factory', () => {
+            // eslint-disable-next-line max-len
+            it('should start existing unarchived branch pr jobs without creating duplicates', () => {
+                jobsMock = [{
+                    id: 1,
+                    pipelineId: 8765,
+                    name: 'main',
+                    permutations: [{
+                        requires: ['~pr:branch']
+                    }],
+                    state: 'ENABLED'
+                }, {
+                    id: 5,
+                    pipelineId: 8765,
+                    name: 'PR-1:main',
+                    permutations: [{
+                        requires: ['~pr:branch']
+                    }],
+                    state: 'ENABLED'
+                }, {
+                    id: 6,
+                    pipelineId: 8765,
+                    name: 'PR-1:outdated',
+                    permutations: [{
+                        requires: ['~pr:branch']
+                    }],
+                    state: 'ENABLED',
+                    archived: true
+                },
+                {
+                    id: 7,
+                    pipelineId: 8765,
+                    name: 'PR-2:main',
+                    permutations: [{
+                        requires: ['~pr:branch']
+                    }],
+                    state: 'ENABLED'
+                },
+                {
+                    id: 8,
+                    pipelineId: 8765,
+                    name: 'pr-branch',
+                    permutations: [{
+                        requires: ['~pr:branch']
+                    }],
+                    state: 'ENABLED'
+                },
+                {
+                    id: 6,
+                    pipelineId: 8765,
+                    name: 'PR-1:pr-branch',
+                    permutations: [{
+                        requires: ['~pr:branch']
+                    }],
+                    state: 'DISABLED'
+                }
+                ];
+
+                afterSyncedPRPipelineMock.getJobs.resolves(jobsMock);
+                afterSyncedPRPipelineMock.getConfiguration = sinon.stub().resolves({
+                    jobs: jobsMock,
+                    workflowGraph: syncedPipelineMock.workflowGraph
+                });
+                afterSyncedPRPipelineMock.update = sinon.stub().resolves(afterSyncedPRPipelineMock);
+                syncedPipelineMock.update = sinon.stub().resolves(syncedPipelineMock);
+
+                config.startFrom = '~pr:branch';
+                config.prRef = 'branch-pr';
+                config.prNum = 1;
+                config.prInfo.ref = 'branch-pr';
+                config.prTitle = 'Update the README with new information';
+                config.webhooks = true;
+
+                return eventFactory.create(config).then((model) => {
+                    assert.instanceOf(model, Event);
+                    assert.notCalled(jobFactoryMock.create);
+                    assert.calledOnce(buildFactoryMock.create);
```

Comment:
`id: 1` and `id: 8` twice, right?

Reply:
`id:1` and `id:8` are not PR job because PR job has prefix `PR-{prNum}:` to job name(like `id:5,6,7,9`). Therefore, those jobs are not started by branch pr event.

Reply:
I got it. lgtm:+1: